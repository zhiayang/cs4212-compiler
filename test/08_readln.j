// 08_readln.j

class Main
{
	Void main()
	{
		Int int;
		Bool bool;
		String string;

		readln(int);
		readln(bool);
		readln(string);

		println("your integer was:");
		println(int);

		println("your boolean was:");
		println(bool);

		println("your string was:");
		println(string);

		println("done");

		new Foo().guess(69);
	}
}

class Foo
{
	Void guess(Int num)
	{
		Int x; Int tries;

		x = 0;
		tries = 0;
		while(x != num)
		{
			readln(x);
			if(x == num) { println("uwu"); }
			else
			{
				if(x < num) { println("higher"); }
				else        { println("lower"); }
			}

			tries = tries + 1;
		}

		println("tries:");
		println(tries);
	}
}
