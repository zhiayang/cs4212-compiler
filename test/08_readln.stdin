69 T hello world
0
500
250
125
62
93
77
69
