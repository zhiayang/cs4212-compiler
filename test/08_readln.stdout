your integer was:
69
your boolean was:
true
your string was:
hello world
done
higher
lower
lower
lower
higher
lower
lower
uwu
tries:
8
